modules/core/include/opencv2/core/cvdef.h

```diff
@@ -505,7 +503,9 @@ CV_INLINE int cvRound( double value )
 /** @overload */
 CV_INLINE int cvRound(float value)
 {
-#if CV_VFP && !defined HAVE_TEGRA_OPTIMIZATION
+#if defined ANDROID && (defined CV_ICC || defined __GNUC__) && defined HAVE_TEGRA_OPTIMIZATION
+    TEGRA_ROUND_FLT(value);
+#elif CV_VFP && !defined HAVE_TEGRA_OPTIMIZATION
```

Comment:
Why the second half? If we do `HAVE_TEGRA_OPTIMIZATION`, but for whatever reason didn't use the O4T optimization of rounding, let's at least use the public optimization.